### DISCUSSION TITLE
Offline calculation vs fs golf

### DISCUSSION THREAD

**Discussion by Wrightn13** [springbok/MLM2PRO-GSPro-Connector#99] [General]
I have the mevo plus but I am using the connector so I can do webcam putting. I'm finding that my offline misses are way on gspro than I am seeing on fsgolf. Like my draws are exaggerated alot. Fs golf says I missed 10 yards left and gspro is more like 20ish yards. The ball flights are accurate but just the amount of curve seems very exaggerated. I know on the flightscope connector there used to be a use fs offline calculation is there a way to do that using this connector?

**Comment by RocknRollGolf** [comment]
Unfortunately no. That is th draw back of this mode of play. In actual fact the GS pro is more accurate . The FS golf app actually under reads. It's something I have discussed with the developers and  so far there is no solution like the offline calculation available

**Reply by Wrightn13** [reply]
Okay gotcha thank you that makes sense. Do you know if there is a way to add more rois into the connector such as aoa club path etc? It's nice to see them in gspro without looking at fs golf.

**Reply by RocknRollGolf** [reply]
Unfortunately no. They are the only metrics available. 
 There is another detailed discussion a couple  of posts back regarding the spin axis offline calculations. If you want to see more info on it.  There was some discussion about having a similar process as the offline calculation in GS pro but as of yet I haven't heard any more about it

**Reply by Wrightn13** [reply]
Okay thank you very much ill go back and take a look. Hopefully maybe in the future it could be added. Thanks again

**Reply by Wrightn13** [reply]
@RocknRollGolf after talking with alot of people on discord and also with gspro support as well. It seems to be the concensus that an issue could be the over reading of spin axis by the mevo plus and that the flighscope offline calculation actual has some type of "smoothing" to the spin axis number. I was thinking is there a way to add some type of smoothing factor to the coding of the connector where when it pulls the spin axis from fs golf it would then add for example a 30 percent reduction to that number? Would it be possible for it to do this on my computer only as a test?

**Reply by RocknRollGolf** [reply]
You would have to talk to @springbok and @wonder99 there are the creators.A similar proposal has been put forward previously. But I think there was some reluctantance as there was some though it may be giving people an advantage . I think it's a good idea considering it's already in play in GS pros inbuilt connector

**Reply by Wrightn13** [reply]
I could definitely see how it could be abused but possibly there is a way to cap it at a value or be able to have it for only mevo plus users. Whatsbthe best way ti get in contact with @springbok and @wonder99 to see if something like this is possible as I feel it would be hugely benif9cial to the mevo plus users.

**Reply by RocknRollGolf** [reply]
You can contact them on here via there github page . You have tagged them so the should respond

**Reply by RocknRollGolf** [reply]
I would like to see that option too . I assume It could be done for mevo users. I had mentioned before with this being open sourced anyone withe the knowledge could alter the connector to add this and gain an advantage with out any one knowing.

**Reply by springbok** [reply]
I'm open to any suggestions about the range of allowable values. As you say the code can be forked and changed to suit. Can you let me know exactly what you would like and I can add it for you, the smaller the correction the better.

**Reply by RocknRollGolf** [reply]
Looking back the comparisons I have done. The average deviation seems to be around 30%. Not sure if your able to make it scalable to have it compensate more as the spin axis increases.  It  will it be a set value across the board.

**Reply by Wrightn13** [reply]
I would agree anywhere between 30 and 40 percent would be a good starting point. I would be more than happy to test it out and compare to see how it works.

**Comment by wonder99** [comment]
I suggest whatever we decide should be fixed (not user-adjustable), so all
mevo players are on equal footing.
Paul

**Reply by Wrightn13** [reply]
I think that would be fair. I wasn't sure if that was a possible thing to do or not.

**Reply by RocknRollGolf** [reply]
Yes . Not user adjustable. In scalable I meant in game calculation not as in user scalable. 
 I am also have to do any beta testing if required. In the GS pro connector it is just a check box to turn it on or off.

**Reply by springbok** [reply]
I haven't kept up with the discussion details so it would help me if you could tell me what you'd like to see. I can add an additional Yes/No field in settings that would only apply if you have set the LM to MEVO+, what would you like this be called, when set 
(Yes) what correction did you want it to perform and on which data item, an example would be helpful.

**Reply by RocknRollGolf** [reply]
I lifted this explanation from another forum. It explains it in a nut shell 


GSPro flightscope offline mode
For those who are not aware. GSPro released a now mode in the flightscope interface called ‘offline mode’.

This improvement takes the reported offline (miss) distance, carry and other parameters to determine the spin axis for the GSPro flight model.

This fix works because it acknowledges that there are differences between the flightscope flight model and GSPro’s, this fix resolves the differences to produce an outcome that aligns to what the device expected to produce in the simulator.

It has greatly improved the flightscope device’s performance in GSPro.

**Reply by RocknRollGolf** [reply]
We have come to the conclusion that a 30% reduction in the offline calculation would achieve a similar outcome with the Springbok connector for mevo+. Hope this makes sense

**Reply by RocknRollGolf** [reply]
I would assum the data iItem would have to be the spin axis . A reduction in spin axis would reduce the side spin been there for reducing the offline calculation issue

**Reply by Wrightn13** [reply]
Yes that's what I was thinking as well seems spin axis is the issue. One person said if a normal launch monitor measures spin axis on a 1-50 scale the mevo plus measures it on a 1-100 scale which makes sense. I'm sure they might adjust a couple other numbers slightly but spin axis is the easiest to fix. I say "easiest" but I'm not sure how much work @springbok and @wonder99 will need to do.

**Reply by RocknRollGolf** [reply]
@Wrightn13 are you on Discord?

**Comment by RocknRollGolf** [comment]
I have also uploaded a instal instructions video to YouTube 
https://youtu.be/Tb1Q-7e_GII?si=ta7RCTeOsZB2e3JZ

**Comment by springbok** [comment]
I've added an offline mode option in settings, default to Yes. It will reduce spin axis by 30%, untested as I don't have a MEVO :) so any issues let me know. 

V1.04.06

https://1drv.ms/u/c/7c92ade387697b0c/EYNxWR_8jspNkPpIrKN1ekYBCVIglScJacs0I2ZsMJdQ1g?e=u1978y

**Reply by Wrightn13** [reply]
Awesome. Thank you very much. I will give it a try tomorrow and let you know what I find. Thank you again.

**Reply by RocknRollGolf** [reply]
@wonder99 sent me a 30% reduction version yesterday. I'm am yet to try it. I will be testing it out tonight and will let you know my results

**Reply by springbok** [reply]
V1.04.05 had a 70% reduction, V1.04.06 sets it to the correct 30%

**Reply by RocknRollGolf** [reply]
Ok thanks for setting that straight. 
So I will run your new  version tonight

**Reply by RocknRollGolf** [reply]
@Wrightn13 @springbok @wonder99 
 So I ran both versions. And believe it or not the 70% reduction was the closest I've ever seen the GS pro and FS golf shot trajectories. I did a bunch of shots on the range with different clubs with both versions. 
I will bundle em in to my one drive and send you link if want to do a comparison. I also played a round with the 70%reduction and the in game ballflight was pretty much identical to FS golf for the entire 18 holes . I can record some holes with both FS golf and GS pro together if you would like to see it.

**Reply by Wrightn13** [reply]
@RocknRollGolf could you provide me with the 1.04.05 with 70 percent reduction also so i could test the with the 1.04.06 with 30 percent also?

**Reply by Wrightn13** [reply]
i just did some testing with the 30 percent reduction and it is so much better erven with that. i do feel like a little more reduction would be great maybe 50 percent or maybe to 70 is the sweet spot. either way to the 30 percent is an amazing start point where i feel more comfortable believing the ball flights in gspro

**Reply by RocknRollGolf** [reply]
@Wrightn13  I can send you a link to it today . Yeah could try with 50-60 % see how that goes

**Reply by RocknRollGolf** [reply]
@Wrightn13 
https://1drv.ms/f/s!Ao9TatRxBybgg5wlPQKCudh_SpQjfw

**Reply by Wrightn13** [reply]
thank you very much ill give that one a try too doing more with the 30 i definetly feel a little more would be great but its so much better than before.

**Reply by RocknRollGolf** [reply]
I did 70% in game test and it was pretty dead on. Could try 50% it might not be enough. 30% is an improvement but its not enough

**Reply by Wrightn13** [reply]
@RocknRollGolf i just opened the 70 percent shared one drive folder you posted above and it saysd its empty could you send it again. thank you

**Reply by springbok** [reply]
https://1drv.ms/u/c/7c92ade387697b0c/ERea_BKayHtOsBlj1DXHDvIBOGIL2AVgFr7z_9Gp4GFovQ?e=fjPL2N

**Comment by springbok** [comment]
So what’s the number you want it set to? If 70% is dead on is that the number or do you want to try 60%.

**Reply by RocknRollGolf** [reply]
Well I'm thinking if we just go increments of 20 we've got a 30 we'll do a 50 and then we've got a 70. If you're happy to send one more at 50% I'm sure  you busy and I don't want to waste your time. The 30% is definitely out so yeah we'll give the 50% a go and make a decision between the 50 and the 70. If that okay

**Comment by springbok** [comment]
V1.04.07 50% adjustment.

https://1drv.ms/u/c/7c92ade387697b0c/EWadOOB-VHhLqAd-7b5T8N8BBm3HT-SSY4FNHCyJ3sAgMQ?e=mOzf3Q

**Reply by RocknRollGolf** [reply]
Thanks I'll get a look at it tomorrow

**Comment by springbok** [comment]
Can you all confirm if 50% is the sweet spot, I want to finalise the changes ASAP so I can do the official release.

**Reply by RocknRollGolf** [reply]
Sorry I have been unable to try it out yet. I should have time tonight to run it

**Reply by Wrightn13** [reply]
I did some small testing yesterday and today and it seems like the 50 percent for me at least was pretty close. Only issue is on huge mis hits which I don't think has anything to do with the connector or anything. But from most of the shots it seems like the offline is within 1 yard or the floghtscope which is so much better. I will do more in depth testing tomorrow as well and will update then. But again thank you so much for the updates it has been very helpful and made the experience so much better

**Reply by RocknRollGolf** [reply]
@Wrightn13 have you tried the 70% yet. 
 I will try the 50%this evening and let you know my thoughts asap @springbok

**Reply by Wrightn13** [reply]
@springbok I have not had the chance for 70 yet. I won't be able to get to it until tomorrow.

**Reply by RocknRollGolf** [reply]
@springbok ok So I have A/B Tested the 50% and 70% tonight, and I believe right in between the two at 60% will be the number 
50 wasn't enough and 70 was too much, So I'm Confidant 60 % will be the number.

**Comment by springbok** [comment]
Ok no worries thanks for testing, I’ll change it to 60 and send a link tomorrow for you to do a final test to confirm it’s the sweet spot.

**Reply by RocknRollGolf** [reply]
No worries. Thanks for work and patience 🙏

**Comment by springbok** [comment]
V1.04.08 using 60% correction:

https://1drv.ms/u/c/7c92ade387697b0c/EfYiJ_Q74FhKva97-JLgMMsBhsMLutpMSQyTaINFu1NKpA?e=6vhgE3

**Reply by RocknRollGolf** [reply]
ok, so 60% still isn't quite there, it would be acceptable, and I know you want to get this finalised, but if you can just make it 65% maybe and leave it at that, we probably don't need to test it anymore. I am just finding that the 70% is closer but it does give a little too much advantage at times. Sorry it's dragging out. I want to get it done and dusted believe me

**Reply by springbok** [reply]
Apologies I think I sent you the wrong file, it was actually a 40% not a 60% correction, this is 60% correction:

 result = float(result * 0.4)

https://1drv.ms/u/c/7c92ade387697b0c/EaJDvc4w0BBLjQxmCmdBzcUBSYFJTIPrLCinIOD5ytbMEQ?e=SeF3gA

**Reply by RocknRollGolf** [reply]
Ok I'm away till Sunday. I'll check it out when I get back.

**Reply by Wrightn13** [reply]
@springbok I tried testing the version 1.04.09  for the 60 percent reduction test and keep getting the error that says exception failed to init API Possibly an invalid tessdata Path i set everything up the same as all the other versions. any suggestions?

**Reply by Wrightn13** [reply]
looking at  the 1.04.09 folder the only thing in it is the .exe MLM2pro connector file and the app date where the previous versions had things such as all the traineddate files. im not sure if this is the issue or if im just looking at it wrong

**Reply by RocknRollGolf** [reply]
ok , I returned today and have spent the afternoon testing the 60% reduction, and can say this is the sweet spot, 
I am good with the version

**Reply by springbok** [reply]
ok great thanks for testing, I'll leave 60% as the level in the released version.

**Comment by springbok** [comment]
You should copy the exe to your existing installation. I didn’t bother including the other files as the only file that changed was the exe.

**Reply by Wrightn13** [reply]
Gotcha. I'll do that and get some testing done also. Thank you.

**Reply by RocknRollGolf** [reply]
If just copy the exe file in you don't have to set up all the ROIs and web cam putting again.

**Reply by springbok** [reply]
> Gotcha. I'll do that and get some testing done also. Thank you.

@Wrightn13 did you get a chance to test the 60% in the last version I posted?

**Reply by Wrightn13** [reply]
I have not yet. I've been away for the weekend. I'm hoping to be able to do some testing tomorrow. Sorry it taking a little longer than the other ones should be able to get some good testing tonorrow then we can get it finalized.

**Reply by springbok** [reply]
No worries no rush, I'm back tomorrow so will look at releasing the next official version late in the week.

**Reply by RocknRollGolf** [reply]
I've already done a bunch of testing on it and I'm using it now and it works spot on.

**Comment by RocknRollGolf** [comment]
I've done a bunch of testing
 I've been using it for a couple of days and it's pretty well bang on now it's all within a meter or two probably on the really worse shots maybe gets out to about 5 m difference with other than that yeah it's great

**Comment by springbok** [comment]
Sometime this week, there should be a link to the release in the discussion thread.
